packages/react/tabs/src/Tabs.tsx

```diff
@@ -176,22 +174,25 @@ const TabsTrigger = React.forwardRef<TabsTriggerElement, TabsTriggerProps>(
           id={triggerId}
           {...triggerProps}
           ref={forwardedRef}
-          // Handle anything that the browser considers a click for the element type if
-          // not using pointer e.g. Space keyup and Enter keydown
-          onClick={composeEventHandlers(props.onClick, handleTabChange)}
           onMouseDown={composeEventHandlers(props.onMouseDown, (event) => {
             // only call handler if it's the left button (mousedown gets triggered by all mouse buttons)
             // but not when the control key is pressed (avoiding MacOS right click)
             if (!disabled && event.button === 0 && event.ctrlKey === false) {
-              handleTabChange();
+              context.onValueChange(value);
             }
           })}
+          onClick={composeEventHandlers(props.onClick, (event) => {
+            // Handle anything that the browser considers a click for the element type if
+            // not using pointer e.g. Space keyup and Enter keydown
+            const isPointerClick = event.detail > 0;
+            if (!isPointerClick) context.onValueChange(value);
+          })}
```

Comment:
I'm really in 2 minds since we've added these `onClick` handlers to all these :(
I get the "why" but I keep having this niggling feeling that ultimately we end up with **more** code and code that is **less explicit** than just implementing the "positive" /explicit cases.

I've felt like that in my Select PR's too where I'm doing this but then having to undo some behaviour anyway.

Am I the only one who feels like this?

Reply:
> I get the "why" but I keep having this niggling feeling that ultimately we end up with more code and code that is less explicit than just implementing the "positive" /explicit cases.

Just to be clear, is your thought that it might be clearer to simply stick with the original explicit keyboard binding over usage of `onClick`?

  Are we functionally gaining anything by using `onClick` in place of? does it help us in any other ways beyond the keyboard stuff?

Reply:
Yes that's what I meant, before introducing `onClick` we were reproducing the behaviour that `onClick` gives in some cases.

But the thing is I feel like over time we realising in multiple components that we don't always want "always all " that behaviour, so we end up guarding against certain conditions, or undoing certain behaviours.

So I feel like when I try to follow the code, it's actually less clear and less explicit in most cases now.

Reply:
Right, yes, I see your point. I guess we need to step back and weigh the benefits to making `onClick` work for us vs the explicit binding stuff. For some reason I thought there was an element of [library interop involved](https://github.com/radix-ui/primitives/issues/980) which `onClick` would make easier. Perhaps @jjenzz has more thoughts.

Reply:
> feeling that ultimately we end up with **more code**

I'm not getting that feeling personally. Perhaps it feels that way because our implementation before didn't match what `onClick` handles? 

> does it help us in any other ways beyond the keyboard stuff?

It mostly just helps because we rely on the browser handling instead of coding it ourselves (incorrectly). For example, we were opening on space **keydown** before, but browsers click buttons on space **keyup** and enter **keydown**. And this behaviour changes per node type, which is worth considering...

If someone changed the node to an anchor, space keypress doesn't click them because that's how the browser decides anchors should behave. So, do we want the behaviour to respect the underlying node type and use `onClick` or do we want to say that it should always open via space keypress regardless of node type?

I can put the following code in instead so we're ensuring the same `button` keyboard behaviour for all node types, but then does that mean we need to add `event.preventDefault()` back in to prevent windows from scrolling for all node types? We agreed we wouldn't add code to support a change in node type so where do we draw the line?

```jsx
onKeyDown={composeEventHandlers(props.onKeyDown, event => {
  if (event.key === 'Enter') context.onOpen();
})}
onKeyUp={composeEventHandlers(props.onKeyUp, event => {
  if (event.key === ' ') context.onOpen();
})}

// vs 
onClick={composeEventHandlers(props.onClick, event => {
  if (event.button === 0) context.onOpen();
})}
```

Reply:
But that's where I'm not sure I agree, because ultimately we are not after a "click" behaviour on a button. If you look at native select, or native dropdown menus in MacOS, for example it's space keydown, and no enter key at all to open. The only reason we added enter key, is because some people asked for it.

So I just feel like somehow we're trying to fit in the native handling of a single button into completely different patterns with a different set of requirements.

Reply:
> If someone changed the node to an anchor, space keypress doesn't click them because that's how the browser decides anchors should behave. So, do we want the behaviour to respect the underlying node type and use onClick or should it always open via space keypress regardless of node type?

I can see this for component that could end up with an anchor trigger, like `HoverCard`, I think yeah that's a valid point. But I don't see this applying to `DropdownMenu` for example, it just wouldn't make sense as part of the pattern.

Reply:
> I can see this for component that could end up with an anchor trigger, like `HoverCard`, I think yeah that's a valid point. But I don't see this applying to `DropdownMenu` for example, it just wouldn't make sense as part of the pattern.

I agree, so perhaps we should evaluate here which components require which approach instead of making it consistent across the board. They have separate concerns.

- `Tooltip` - `onClick` so it closes when button or anchor is considered "clicked"
- `DropdownMenu` - `onKeyDown` so it behaves same for all node types
-  `Tabs` - ?

Reply:
Yeah it seems we should handle them one by one for what they are, I agree.

> Also, what about the event.preventDefault() question?

You mean for preventing to scroll when pressing space?
I guess same, we should only do it when it's necessary?

Reply:
> I guess same, we should only do it when it's necessary?

When is it necessary though? 

Right now it's not necessary in any of our components because the triggers are `button`s but if someone changes them to an `a` or a`<div role="button">` then it will be necessary. 

I recall we decided against adding code to support changes to the underlying node though so do we just leave it out?

Reply:
I think is the use is normally a button then we shouldn't do anything (unless it's necessary to be patched in some browsers). We should only consider doing it for primitives where the trigger may not be a button I suppose.

Reply:
Sweet, I'll mark this as resolved then. I've gone with `onKeyDown` for `Tabs`.